crates/apub/src/lib.rs

```diff
@@ -97,36 +98,47 @@ pub(crate) struct LocalSiteData {
   blocked_instances: Vec<Instance>,
 }
 
-pub(crate) async fn fetch_local_site_data(
-  pool: &DbPool,
-) -> Result<LocalSiteData, diesel::result::Error> {
-  // LocalSite may be missing
-  let local_site = LocalSite::read(pool).await.ok();
-  let allowed_instances = Instance::allowlist(pool).await?;
-  let blocked_instances = Instance::blocklist(pool).await?;
-
-  Ok(LocalSiteData {
-    local_site,
-    allowed_instances,
-    blocked_instances,
-  })
+pub(crate) async fn local_site_data_cached(pool: &DbPool) -> LemmyResult<Arc<LocalSiteData>> {
```

Comment:
This means that any updates to the site aren't going to be picked up for 30 minutes. That means these things like this will be out of date:

- Someone adds an instance to the blocklist.
- Someone updates the site description.

Reply:
This is only used for blocklist/allowlist checks and for slur checks. Not for any API responses. But yes I will make the interval shorter.